src/plugins/data/public/query/query_string/query_string_manager.ts

```diff
@@ -225,7 +225,28 @@ export class QueryStringManager {
    * Resets the query to the default one.
    */
   public clearQuery = () => {
-    this.setQuery(this.getDefaultQuery());
+    const curQuery = this.query$.getValue();
```

Comment:
this is doing awful lot for a method called `clearQuery()`. Looking at the code, it looks like we gotrta handle the scenario where dataset has changed. would it be feasible to have this code in like a new method called `handleDatasetChange` or something?